config/jobs/kubernetes/sig-network/dualstack-e2e.yaml

```diff
@@ -203,3 +203,56 @@ periodics:
     testgrid-tab-name: dualstack-azure-e2e-master
     testgrid-alert-email: kubernetes-provider-azure@googlegroups.com
     description: "Dual-stack e2e tests on a Kubernetes cluster provided by aks-engine (https://github.com/Azure/aks-engine) on Azure cloud"
+
+presubmits:
+  kubernetes/kubernetes:
+  - name: pull-kubernetes-e2e-azure-dualstack
+    decorate: true
+    decoration_config:
+      timeout: 2h
+    always_run: false
+    optional: true
+    path_alias: k8s.io/kubernetes
+    branches:
+    - master
+    labels:
+      preset-service-account: "true"
+      preset-azure-cred: "true"
+      preset-dind-enabled: "true"
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20201005-0b243c0-master
+        command:
+        - runner.sh
+        - kubetest
+        args:
+        # Generic e2e test args
+        - --test
+        - --up
+        - --down
+        - --build=quick
+        - --dump=$(ARTIFACTS)
+        # Azure-specific test args
+        - --provider=skeleton
+        - --deployment=aksengine
+        - --aksengine-agentpoolcount=2
+        - --aksengine-admin-username=azureuser
+        - --aksengine-creds=$(AZURE_CREDENTIALS)
+        - --aksengine-orchestratorRelease=1.19
+        - --aksengine-mastervmsize=Standard_DS2_v2
+        - --aksengine-agentvmsize=Standard_D4s_v3
+        - --aksengine-deploy-custom-k8s
+        - --aksengine-location=westus2
+        - --aksengine-public-key=$(AZURE_SSH_PUBLIC_KEY_FILE)
+        - --aksengine-template-url=https://gist.githubusercontent.com/aramase/21c3bdb44509c0f70f71389903d49c67/raw/1a9957626edc2b5c7913c741b9487fde123d0570/dualstack-prow.json
+        - --aksengine-download-url=https://aka.ms/aks-engine/aks-engine-k8s-e2e.tar.gz
+        # Specific test args
+        - --test_args=--ginkgo.focus=\[Feature:IPv6DualStackAlphaFeature\]
```

Comment:
new tests have this tag `[Feature:IPv6DualStackAlphaFeature:Phase2]` , seems we have to use it without the trailing square bracket :shrug: 
```suggestion
        - --test_args=--ginkgo.focus=\[Feature:IPv6DualStackAlphaFeature
```

Reply:
actually, we should run more tests to be sure that all features works, this is very happy scenario

Reply:
once we verify this works, can you do the same changes for the periodic in a follow-up PR?

Reply:
>new tests have this tag [Feature:IPv6DualStackAlphaFeature:Phase2] , seems we have to use it without the trailing square bracket 🤷

So phase2 tests run too with `[Feature:IPv6DualStackAlphaFeature\]`- https://testgrid.k8s.io/provider-azure-dualstack#dualstack-azure-e2e-master.

Reply:
Yeah, +1 for running conformance. The periodic jobs initial focus was to just validate the dualstack e2e tests run as expected. We can start running it for presubmit and then port back the change to periodic.

Reply:
yeah, but checks Khals PR 
https://github.com/kubernetes/kubernetes/pull/91824/commits/3d7ebf0fd96d2f5b78e77aba6bde2f4f074a2ecd#diff-62557e1c28188e811e7529d548cacb8cR352

we should change it here or there, can you both discuss it an agree on a solution?

Reply:
@aojea So this regexp matches `Feature:IPv6DualStackAlphaFeature` and `Feature:IPv6DualStackAlphaFeature:Phase2`. I tried that out locally.

Reply:
that is really weird, isn't it?

Reply:
![image](https://user-images.githubusercontent.com/6450081/95498991-eb2fd880-09a4-11eb-8c9b-5a397a6a99fd.png)

Reply:
yeah, this is weird. 

```
➜ _output/local/bin/linux/amd64/e2e.test -ginkgo.focus="Feature:IPv6DualStackAlphaFeature" -ginkgo.dryRun
Ran 9 of 5228 Specs in 0.027 seconds
SUCCESS! -- 0 Passed | 0 Failed | 0 Pending | 5219 Skipped
PASS
```
Also see  https://testgrid.k8s.io/provider-azure-dualstack#dualstack-azure-e2e-master. This runs few existing phase2 tests too with the same regex.

Reply:
your regex does not have the square brackets, you have to add those 
--ginkgo.focus=\[Feature:IPv6DualStackAlphaFeature\] 

this way I think it will not match

Reply:
@aojea But that's the same regex used in the periodic job. That's running phase2 tests. 

```
[sig-network] [Feature:IPv6DualStackAlphaFeature] [LinuxOnly] should be able to reach pod on ipv4 and ipv6 ip [Feature:IPv6DualStackAlphaFeature:Phase2]
[sig-network] [Feature:IPv6DualStackAlphaFeature] [LinuxOnly] should create pod, add ipv6 and ipv4 ip to pod ips
[sig-network] [Feature:IPv6DualStackAlphaFeature] [LinuxOnly] should create service with cluster ip from primary service range [Feature:IPv6DualStackAlphaFeature:Phase2]
[sig-network] [Feature:IPv6DualStackAlphaFeature] [LinuxOnly] should create service with ipv4 cluster ip [Feature:IPv6DualStackAlphaFeature:Phase2]
[sig-network] [Feature:IPv6DualStackAlphaFeature] [LinuxOnly] should create service with ipv6 cluster ip [Feature:IPv6DualStackAlphaFeature:Phase2]
[sig-network] [Feature:IPv6DualStackAlphaFeature] [LinuxOnly] should have ipv4 and ipv6 internal node ip
[sig-network] [Feature:IPv6DualStackAlphaFeature] [LinuxOnly] should have ipv4 and ipv6 node podCIDRs
```

Reply:
duh, sorry @aramase this means I have to stop for today, 
```
[sig-network] [Feature:IPv6DualStackAlphaFeature] [LinuxOnly] should create service with ipv6 cluster ip [Feature:IPv6DualStackAlphaFeature:Phase2]
```
it matches in the first tag